CDL-Dryad/dryad-app
# Utilities for connecting with the journal email account

For https://github.com/CDL-Dryad/dryad-product-roadmap/issues/703

Adds a utility class to facilitate processing emails from Dryad's partner journals.

Before working with the journal email account, Rails must obtain a token that allows access to that account. The token can be obtained by running `rails journal_email:validate_gmail_connection` and following the listed directions. (Note that this requires access to the GMail account for journal-submit-app@datadryad.org)

There is only one journal email account, which receives both test messages and real messages from journals. It is used by the production email processing system (currently still on the v1 server), so it is best to avoid deleting any messages from this account. Testing can be performed by adding and removing the email label that triggers processing. It is `dev-journal-submit` for all non-production servers.

To test: 
1. Create/verify the token with `rails journal_email:validate_gmail_connection`
2. Ensure at least one message in the account has the label `dev-journal-submit`
3. In a Rails console session:
```
require 'stash/google/journal_gmail'
m=Stash::Google::JournalGMail.messages_to_process.first
mc=Stash::Google::JournalGMail.message_contents(message: m)
```
